### PR TITLE
Add User API

### PR DESCRIPTION
**Pull request by AhmedTammam**
- delete mockgoose library
- rename files according to its roles
- build auth router and controller
- write user tests

### CODE REVIEW

File: controllers/auth-controller.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const User = require("../models/user");
+const User = require("../models/user-model");
```
Comment (xfoxawy) [inline controllers/auth-controller.js:1]: a bit redundant to have `models/user-model`
I think its fluent `models/user`

File: controllers/auth-controller.js [hunk 2]
```diff
@@ -54,3 +54,23 @@ exports.requireSignin = expressJwt({
...
+  if (!user) {
+    return res.status(403).json({
+      error: "Access denied",
+    });
+  }
+  next();
+};
+
+exports.isAdmin = (req, res, next) => {
+  if (req.profile.role === 0) {
```
Comment (xfoxawy) [inline controllers/auth-controller.js:70]: replace comparison to const not zero
```
 req.profile.role == Roles.Admin
```
more readable

File: controllers/user-controller.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const User = require("../models/user-model");
+
+exports.findUserById = (req, res, next, id) => {
```
Comment (xfoxawy) [inline controllers/user-controller.js:3]: this should live inside a repo not a controller

Comment (AhmedTammam) [reply controllers/user-controller.js]: > this should live inside a repo not a controller

what you mean with repo ?? 😃 😃

**Pull request merged**